src/constants/tokens.ts

```diff
@@ -4,27 +4,19 @@ import { Erc20CoinData } from './types/Erc20CoinData';
 
 export const getErc20Tokens = (parent: CoinData) => {
   const TOKENSLIST: Record<string, Erc20CoinData> = {};
-  let tokensList: any[];
-
-  switch (parent.id) {
-    case 'ethereum':
-      tokensList = erc20List;
-      break;
-    default:
-      tokensList = [];
-  }
+  const tokensList: any = erc20List;
 
   for (const token of tokensList) {
-    if (token.symbol.length <= 16) {
+    if (token.symbol.length <= 16 && token.platforms[parent.id]) {
       const id = `${parent.id}:${token.id}`;
       TOKENSLIST[id] = new Erc20CoinData({
         id,
         parentId: parent.id,
         oldId: token.symbol.toLowerCase(),
         abbr: token.symbol.toLowerCase(),
         coinGeckoId: token.id,
-        address: token.address,
-        decimal: token.decimal ?? 18,
+        address: token.platforms[parent.id].contract_address,
+        decimal: token.platforms[parent.id].decimal_place ?? 18,
```

Comment:
What does it mean when `decimal_place` is undefined on the JSON file?

Can we make sure the `decimal_place` is present on the JSON file and remove the default case?

Reply:
That is already the case, the JSON file will contain decimal_place. Just left the fallback there, as it shows we are using 18 as the default case (even though this condition won't ideally run)

Reply:
If the fallback is not meant to run, then throw an error if there is no `decimal_place` or `contract_address`. This will make sure that any bug in the JSON file is caught during testing.